subsys/net/ip/net_context.c

```diff
@@ -642,6 +642,10 @@ struct net_pkt *net_context_create_ipv4(struct net_context *context,
 	    || net_ipv4_is_addr_mcast(src)) {
 		src = net_if_ipv4_select_src_addr(net_pkt_iface(pkt),
 						  (struct in_addr *)dst);
+		/* If src address is still unspecified, do not create pkt */
+		if (net_ipv4_is_addr_unspecified(src)) {
+			return NULL;
```

Comment:
Here's where I suggest to add debug logging. Unless you can point to to the fact that net_if_ipv4_select_src_addr() call above already does *clear* logging (like, it hints that if it fails, then some packets can be not sent out, which I guess it doesn't).